### PR TITLE
Improve map tests

### PR DESCRIPTION
**Pull request by gdbelvin**
Make the map test more robust by not referencing nil pointers

### CODE REVIEW

File: integration/maptest/map.go [hunk 1]
```diff
@@ -177,55 +177,58 @@ func RunMapRevisionZero(ctx context.Context, t *testing.T, tadmin trillian.Trill
...
+					MapId: tree.TreeId,
+				})
+				if err != nil {
+					t.Fatalf("GetSignedMapRoot(): %v", err)
+				}
+				if err := verifyGetSignedMapRootResponse(getSmrResp.GetMapRoot(), tc.wantRev,
+					pubKey, hasher, tree.TreeId); err != nil {
+					t.Errorf("verifyGetSignedMapRootResponse(rev %v): %v", tc.wantRev, err)
+				}
+				//
```
Comment (AlCutter) [inline integration/maptest/map.go]: What are these blank comment lines for, would a blank line suffice?

Comment (gdbelvin) [reply integration/maptest/map.go]: Not sure. I'll remove it. I think it was pre-existing.

**Comment by gdbelvin** [comment]
PTAL

File: integration/maptest/map.go [hunk 2]
```diff
@@ -177,55 +177,57 @@ func RunMapRevisionZero(ctx context.Context, t *testing.T, tadmin trillian.Trill
...
-				t.Errorf("%v: newTreeWithHasher() returned unexpected root hash %x, want %x", tc.desc, got, want)
-			}
-			pubKey, err := der.UnmarshalPublicKey(tree.GetPublicKey().GetDer())
-			if err != nil {
-				t.Errorf("%v: UnmarshalPublicKey(%v): %v", tc.desc, hsr.hashStrategy, err)
-			}
+			t.Run(fmt.Sprintf("%v/%v", tc.desc, hsr.hashStrategy), func(t *testing.T) {
+				tree, hasher, smr, err := newTreeWithHasher(ctx, tadmin, tmap, hsr.hashStrategy)
+				if err != nil {
+					t.Fatalf("newTreeWithHasher(%v): %v", hsr.hashStrategy, err)
```
Comment (phad) [inline integration/maptest/map.go]: Does t.Fatalf() kill the entire test suite or will it be recovered by test.T so the next sub-test can run?

Comment (gdbelvin) [reply integration/maptest/map.go]: Fatalf only kills the test currently inside `t.Run` which is nice.

**Pull request merged**